retrieval/discovery/marathon/url.go

```diff
@@ -0,0 +1,14 @@
+package marathon
+
+import (
+	"fmt"
+	"math/rand"
+)
+
+const appsURLFormat string = "http://%s/v2/apps/?embed=apps.tasks"
+
+// RandomAppsURL randomly selects a server from an array and creates an URL pointing to the app list
+func RandomAppsURL(servers []string) string {
```

Comment:
The problem here is that we are forever bound to the list in the configuration. Thanks for config reloading this is kind of okay. Frequently syncing all available servers is more reliable, though. 
Do I see it correctly that the current API doesn't provide a list of all peers but only the current leader?

If so, please add a TODO explaining the situation in the hope that the Marathon API will be extended by this feature.